yt/data_objects/profiles.py

```diff
@@ -416,18 +416,20 @@ class ProfileNDFromDataset(ProfileND):
     """
     def __init__(self, ds):
         ProfileND.__init__(self, ds.data, ds.parameters["weight_field"])
-        self.fractional = ds.parameters["fractional"]
-        self.accumulation = ds.parameters["accumulation"]
+        self.fractional = ds.parameters.get("fractional", False)
+        self.accumulation = ds.parameters.get("accumulation", False)
         exclude_fields = ["used", "weight"]
         for ax in "xyz"[:ds.dimensionality]:
             setattr(self, ax, ds.data[ax])
-            setattr(self, "%s_bins" % ax, ds.data["%s_bins" % ax])
-            field_name = tuple(ds.parameters["%s_field" % ax])
-            setattr(self, "%s_field" % ax, field_name)
+            ax_bins = "%s_bins" % ax
+            ax_field = "%s_field" % ax
+            ax_log = "%s_log" % ax
+            setattr(self, ax_bins, ds.data[ax_bins])
+            field_name = tuple(ds.parameters[ax_field])
```

Comment:
Does it make sense to have a default value here too or are `x_field` `y_field` and `z_field` always in `ds.parameters`?

Reply:
I think this is a more reliable assumption, but I agree it's probably better not to error out with a `KeyError` in the unlikely event. I'll convert that to a get, too.